### ISSUE TITLE
Progress bar overflows on smaller terminals

### ISSUE THREAD

**Issue by Dorian** [curl/curl#2113]
### I did this

```
curl --progress-bar https://www.openssl.org/source/openssl-1.0.2m.tar.gz -o tmp
###                                                                   ######                                                                ###########                                                           ###############                                                       ##################                                                    ######################                                                ##########################                                            #############################                                         #################################                                     #####################################                                 ########################################                              ############################################                          ################################################                      ###################################################                   #######################################################               ###########################################################           ##############################################################        ##################################################################    ############################################################################################################################################## 100.0%
```

<img width="300" alt="screen shot 2017-11-26 at 10 55 47 am" src="https://user-images.githubusercontent.com/129149/33238957-6a05a334-d298-11e7-9bba-1f13716c603a.png">

### I expected the following

`############################################################### 100.0%`

63 "#" and " 100.0%" (my terminal is 70 characters long)

e.g. `env COLUMNS=70 curl --progress-bar https://www.openssl.org/source/openssl-1.0.2m.tar.gz -o tmp` works. So maybe it's a bug in `curlx_getenv`.

### curl/libcurl version

```
curl 7.43.0 (x86_64-apple-darwin14.0) libcurl/7.43.0 SecureTransport zlib/1.2.5
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: AsynchDNS IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz UnixSockets
```

### operating system

`macOS 10.10.5 (14F2411)`

Happens with fish shell and bash in Terminal and iTerm

**Comment by Dorian** [comment]
Seems to be the hardcoded `79` in https://github.com/curl/curl/blob/master/src/tool_cb_prg.c#L133-L137.

But I have `$COLUMNS` set to 70.

**Comment by Dorian** [comment]
Also this TODO seems to be what I expected:

```
TODO: get terminal width through ansi escapes or something similar.
   try to update width when xterm is resized... - 19990617 larsa
```

**Comment by Dorian** [comment]
Also, `getenv('COLUMNS')` doesn't work, it needs to be done with `ioctl`, e.g. https://stackoverflow.com/a/1022961/407213

**Comment by bagder** [comment]
> Also, getenv('COLUMNS') doesn't work, it needs to be done

I tested on both Linux and mac just now and setting COLUMNS works just fine for me...

**Comment by bagder** [comment]
Fixed in git since commit 993dd5651a
